databricks/koalas/namespace.py

```diff
@@ -238,11 +240,23 @@ def read_csv(path, header='infer', names=None, usecols=None,
                 sdf = default_session().createDataFrame([], schema=StructType())
     else:
         sdf = default_session().createDataFrame([], schema=StructType())
-    return DataFrame(sdf)
+
+    index_map = None  # type: Optional[List[IndexMap]]
+
+    if index_col is not None:
+        if isinstance(index_col, str):
+            index_col = [index_col]
+        sdf_columns = set(sdf.columns)
+        for col in index_col:
+            if col not in sdf_columns:
+                raise KeyError(col)
+        index_map = [(col, col) for col in index_col]
```

Comment:
We should extract this to a function to share the code?

Reply:
oh i think it's a better way either. I've extract it as a `_get_inde_map` function :)